utils/src/foreign_field.rs

```diff
@@ -413,33 +449,55 @@ mod tests {
         let big = secp256k1_modulus();
         let bytes = big.to_bytes_be();
         assert_eq!(
-            ForeignElement::<BaseField, 3>::from_be(&bytes),
-            ForeignElement::<BaseField, 3>::from_biguint(big)
+            ForeignElement::<BaseField, LIMB_BITS, 3>::from_be(&bytes),
```

Comment:
(we cannot comment on unchanged lines)
Not for this PR, but I would use a randomized seed, see above:
```
    const RNG_SEED: [u8; 32] = [
        12, 31, 143, 75, 29, 255, 206, 26, 67, 193, 86, 160, 1, 90, 131, 221, 86, 168, 4, 95, 50,
        48, 89, 29, 13, 250, 215, 172, 130, 24, 164, 162,
    ];
```

It is a small change I would suggest to do as a quick follow-up after.

Reply:
Hmmm I think it's actually good to do it in this PR too to make sure that the tests /actually/ work and nothing breaks?

Reply:
I'm not against :)

Reply:
I did it here: https://github.com/o1-labs/proof-systems/pull/1942